src/output_gstreamer.c

```diff
@@ -44,111 +44,83 @@
-		    ((padtemplate->presence == GST_PAD_ALWAYS) ||
-		     (padtemplate->presence == GST_PAD_SOMETIMES) ||
-		     (padtemplate->presence == GST_PAD_REQUEST)) &&
-		    (padtemplate->caps)) {
-			scan_caps(padtemplate->caps);
-		}
-	}
+    // Ignore elements without pads
+    if (gst_element_factory_get_num_pad_templates(factory) == 0) continue;
 
-}
+    // Fetch a list of all pads
+    const GList* pads = gst_element_factory_get_static_pad_templates(factory);
 
+    while (pads) {
+      GstStaticPadTemplate* padTemplate = (GstStaticPadTemplate*)pads->data;
 
-static void scan_mime_list(void)
-{
-	GstRegistry *registry = NULL;
-	GList *plugins = NULL;
+      // Advance list
+      pads = g_list_next(pads);
 
-#if (GST_VERSION_MAJOR < 1)
-	registry = gst_registry_get_default();
-	plugins = gst_default_registry_get_plugin_list();
-#else
-	registry = gst_registry_get();
-	plugins = gst_registry_get_plugin_list(registry);
-#endif
+      // Skip pads that aren't sinks
+      if (padTemplate->direction != GST_PAD_SINK) continue;
 
-	while (plugins) {
-		GList *features;
-		GstPlugin *plugin;
-
-		plugin = (GstPlugin *) (plugins->data);
-		plugins = g_list_next(plugins);
-
-		features =
-			gst_registry_get_feature_list_by_plugin(registry,
-							    gst_plugin_get_name
-							    (plugin));
-
-		while (features) {
-			GstPluginFeature *feature;
-
-			feature = GST_PLUGIN_FEATURE(features->data);
-
-			if (GST_IS_ELEMENT_FACTORY(feature)) {
-				GstElementFactory *factory;
-				GstElement *element;
-				factory = GST_ELEMENT_FACTORY(feature);
-				element =
-				    gst_element_factory_create(factory, NULL);
-				if (element) {
-					scan_pad_templates_info(element,
-								factory);
-				}
-			}
+      // This is literally all known pad presences so it should be OK!
+      assert(padTemplate->presence == GST_PAD_ALWAYS ||
+             padTemplate->presence == GST_PAD_SOMETIMES ||
+             padTemplate->presence == GST_PAD_REQUEST);
 
-			features = g_list_next(features);
-		}
-	}
-}
+      GstCaps* capabilities = gst_static_caps_get(&padTemplate->static_caps);
+
+      // Skip capabilities that they tell us nothing
+      if (capabilities == NULL || gst_caps_is_any(capabilities) ||
+          gst_caps_is_empty(capabilities))
+        {
+          gst_caps_unref(capabilities);
+          continue;
+        }
 
+      for (guint i = 0; i < gst_caps_get_size(capabilities); i++) {
+        GstStructure* structure = gst_caps_get_structure(capabilities, i);
+
+        register_mime_type(gst_structure_get_name(structure));
+      }
+
+      gst_caps_unref(capabilities);
+    }
+  }
+
+	// Free any allocated memory
+  gst_plugin_feature_list_free(root);
+
+	// There seem to be all kinds of mime types out there that start with
+	// "audio/" but are not explicitly supported by gstreamer. Let's just
+	// tell the controller that we can handle everything "audio/*" and hope
+	// for the best.
+	register_mime_type("audio/*");
```

Comment:
The indentation of this line looks funny compared to the one before.

Reply:
Updated all the idents to be tab-based which was the existing indentation format in the file (mostly).